File: app/src/main/java/foodie/app/rubikkube/foodie/model/CommentData.java
package foodie.app.rubikkube.foodie.model;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class CommentData {
    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("post_id")
    @Expose
    private Integer postId;
    @SerializedName("user_id")
    @Expose
    private Integer userId;
    @SerializedName("content")
    @Expose
    private String content;
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("status")
    @Expose
    private Object status;
    @SerializedName("created_at")
    @Expose
    private String createdAt;
    @SerializedName("updated_at")
    @Expose
    private String updatedAt;
    @SerializedName("user")
    @Expose
    private User user;

    public Integer getId() { return id; }

    public void setId(Integer id) { this.id = id; }

    public Integer getPostId() { return postId; }

    public void setPostId(Integer postId) { this.postId = postId; }

    public Integer getUserId() { return userId; }

    public void setUserId(Integer userId) { this.userId = userId; }

    public String getContent() { return content; }

    public void setContent(String content) { this.content = content; }

    public String getType() { return type; }

    public void setType(String type) { this.type = type; }

    public Object getStatus() { return status; }

    public void setStatus(Object status) { this.status = status; }

    public String getCreatedAt() { return createdAt; }

    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }

    public String getUpdatedAt() { return updatedAt; }

    public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }

    public User getUser() { return user; }

    public void setUser(User user) { this.user = user; }
}

File: app/src/main/java/foodie/app/rubikkube/foodie/models/CommentData.kt
package foodie.app.rubikkube.foodie.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class CommentData(
    @Expose
    @SerializedName("id")
    var id: Int? = null,

    @Expose
    @SerializedName("post_id")
    var postId: Int? = null,

    @SerializedName("user_id")
    @Expose
    var userId: Int? = null,

    @Expose
    @SerializedName("content")
    var content: String? = null,

    @Expose
    @SerializedName("type")
    var type: String? = null,

    @Expose
    @SerializedName("status")
    var status: String? = null,

    @Expose
    @SerializedName("created_at")
    var createdAt: String? = null,

    @Expose
    @SerializedName("updated_at")
    var updatedAt: String? = null,

    @Expose
    @SerializedName("user")
    var user: User? = null
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readValue(Int::class.java.classLoader) as? Int,
        parcel.readValue(Int::class.java.classLoader) as? Int,
        parcel.readValue(Int::class.java.classLoader) as? Int,
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readParcelable(User::class.java.classLoader)
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(id)
        parcel.writeValue(postId)
        parcel.writeValue(userId)
        parcel.writeString(content)
        parcel.writeString(type)
        parcel.writeString(status)
        parcel.writeString(createdAt)
        parcel.writeString(updatedAt)
        parcel.writeParcelable(user, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<CommentData> {
        override fun createFromParcel(parcel: Parcel): CommentData {
            return CommentData(parcel)
        }

        override fun newArray(size: Int): Array<CommentData?> {
            return arrayOfNulls(size)
        }
    }
}